com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/parser/ISemanticPredicates.java

```diff
@@ -62,6 +64,22 @@ public String getMessage(final String key, final Token token) {
       }
       return message;
     }
-  }
 
+    /**
+     * Predicate for grammar rule DefaultFieldFormat.
+     *
+     * @param parserContext
+     *          Input from Lexer
+     * @param firstKeyword
+     *          the current keyword parser is looking at
+     * @param secondKeyword
+     *          the keyword following the current keyword
+     * @return {@code true} if the grammar rule is enabled, {@code false} otherwise
+     */
+    public static boolean isFollowedBy(final ParserContext parserContext, final String firstKeyword, final String secondKeyword) {
+      String current = parserContext.getInput().LT(1).getText();
+      String next = parserContext.getInput().LT(2).getText();
+      return firstKeyword.equalsIgnoreCase(current) && secondKeyword.equalsIgnoreCase(next);

```

Comment:
Could we use lazy evaluation with this version?

      TokenStream stream = parserContext.getInput();
      return firstKeyword.equalsIgnoreCase(stream.LT(1).getText()) && secondKeyword.equalsIgnoreCase(stream.LT(2).getText());